editors/code/src/main.ts

```diff
@@ -269,7 +269,8 @@ function serverPath(config: Config): string | null {
 async function isNixOs(): Promise<boolean> {
     try {
         const contents = (await vscode.workspace.fs.readFile(vscode.Uri.file("/etc/os-release"))).toString();
-        return contents.indexOf("ID=nixos") !== -1;
+        const idString = contents.split('\n').find((a) => a.startsWith("ID="));
+        return idString?.toLowerCase()?.indexOf("nixos") !== -1;
```

Comment:
I think a `contents.indexOf("nixos") !== -1` would have worked just as well (while being less allocation-heavy), but oh well.

Reply:
just in case they decide to do strange stuff with ids
you are probably right that won't be ever necessary